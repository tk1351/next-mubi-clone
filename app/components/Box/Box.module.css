.box {
    padding: 10px;
    border: 0 solid;
    outline: 0 solid;
    outline-offset: 0;
    color: #000;
    background-color: #fff;
}

.box * {
    color: inherit;
}
