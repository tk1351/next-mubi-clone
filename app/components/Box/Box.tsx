import { type FC, type PropsWithChildren } from "react";
import styles from "./Box.module.css";

export type Props = PropsWithChildren<{}>;

export const Box: FC<Props> = ({ children }) => {
  return <div className={styles.box}>{children}</div>;
};
