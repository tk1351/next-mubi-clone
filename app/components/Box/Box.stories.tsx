import type { Meta, StoryObj } from "@storybook/react";
import { Box } from "./Box";

const meta = {
  title: "Components/Box",
  component: Box,
} satisfies Meta<typeof Box>;

export default meta;
type Story = StoryObj<typeof meta>;
export const Primary: Story = {
  args: {
    children: (
      <div
        style={{ height: "100%", width: "100%", border: "1px solid purple" }}
      >
        <h3>Card</h3>
        <p>Description</p>
      </div>
    ),
  },
};

export const Nested: Story = {
  args: {
    children: (
      <>
        <Box>
          <header
            style={{
              height: "100%",
              width: "100%",
              border: "1px solid orange",
            }}
          >
            Header
          </header>
        </Box>
        <Box>
          <div
            style={{
              height: "100%",
              width: "100%",
              border: "1px solid purple",
            }}
          >
            <h3>Card</h3>
            <p>Description</p>
          </div>
        </Box>
      </>
    ),
  },
};
