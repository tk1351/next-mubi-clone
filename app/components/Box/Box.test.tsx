import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Box } from "./Box";

describe(`Testing Box Component`, () => {
  it("Snapshot", () => {
    const { container } = render(<Box>{"Box"}</Box>);
    expect(container).toMatchSnapshot();
  });
});
