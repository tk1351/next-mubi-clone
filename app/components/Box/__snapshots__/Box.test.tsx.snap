// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Box Component > Snapshot 1`] = `
<div>
  <div
    class="_box_61dfa1"
  >
    Box
  </div>
</div>
`;
